examples/test1.rs

```diff
@@ -16,7 +16,7 @@ fn ping(from: &Machine, to: &Machine, log: &slog::Logger) -> Result<(), failure:
 }
 
 fn main() -> Result<(), failure::Error> {
-    let mut b = TsunamiBuilder::<aws::AWSRegion>::default();
+    let mut b = TsunamiBuilder::<aws::AWSLauncher<_>>::default();
```

Comment:
I don't think you need this type annotation any more. In fact, I don't think `TsunamiBuilder` needs to be generic over `L` at all. Instead, you can just have `L: Launcher` be a parameter on `spawn`  I think.

Reply:
`TsunamiBuilder` still needs to be`<L: Launcher>` because it internally holds `HashMap<String, L::Machine>`, the provider-specific setup structs.

Reply:
I'm not sure I follow. Where are those provider-specified entries created?

Reply:
https://github.com/jonhoo/tsunami/pull/22/files/23e32b6cc39d9a9280c19d016b724e3b9f3bc9a1#diff-8a21d79163884c379283e8dcfc728f67R27

The user instantiates an `L::Machine` and passes it to `add(&mut self, nickname: &str, m: L::Machine)`, which is where the type parameter comes in.

I agree the type annotation can be removed, but the type parameter still needs to be there?

Reply:
I think I would make the builder just generic over the machine type `M` then, and then make `spawn`:
```rust
fn spawn<L: Launcher<Machine = M>>(...)
```

Reply:
Seems reasonable, done